zerver/views/user_settings.py

```diff
@@ -357,6 +374,84 @@ def json_change_settings(
     # TODO: Change the cache flushing strategy to make sure cache
     # does not contain stale objects.
     user_profile = UserProfile.objects.get(id=user_profile.id)
+
+    # Loop over user_profile.property_types
+    request_settings = {
+        setting_name: requested_value
+        for setting_name, requested_value in locals().items()
+        if setting_name in user_profile.property_types and requested_value is not None
+    }
+
+    if target_users is not None:
+        if not user_profile.is_realm_admin:
+            raise JsonableError(
+                _("Only organization administrators can change settings for other users.")
+            )
+
+        if timezone is not None:
```

Comment:
I am not sure about this but this is what I understand from the comment on previous PR that we would just not allow to update `timezone` by admins.